webapp/src/Controller/BaseController.php

```diff
@@ -219,21 +219,45 @@ protected function commitDeleteEntity(
         // Now actually delete the entity.
         $entityManager->wrapInTransaction(function () use ($entityManager, $entity) {
             if ($entity instanceof Problem) {
-                // Deleting a problem is a special case: its dependent tables do not
-                // form a tree, and the deletion of judging_run can only cascade from
-                // judging, not from testcase. Since MySQL does not define the
-                // order of cascading deletes, we need to manually first cascade
-                // via submission -> judging -> judging_run.
+                // Deleting a problem is a special case:
+                // Its dependent tables do not form a tree (but something like a diamond shape),
+                // and there are multiple cascading removal paths from problem to its dependant
+                // tables.
+                // Since MySQL does not define the order of cascading deletes, we need to manually
```

Comment:
Remove duplicate `manually`:
```suggestion
                // Since MySQL does not define the order of cascading deletes, we need to
```